Makefile

```diff
@@ -163,14 +163,25 @@ endif
 
 # Targets and their dependencies on externals
 TARGETS := $(notdir $(wildcard $(SRC_DIR)/*))
-TEST_CPU_TARGETS := $(patsubst %,test_%_cpu,$(TARGETS))
-TEST_CUDA_TARGETS := $(patsubst %,test_%_cuda,$(TARGETS))
+TARGETS_SYCL = sycltest
+TARGETS_GCC_NVCC = $(filter-out $(TARGETS_SYCL),$(TARGETS))
+
+TEST_CPU_TARGETS := $(patsubst %,test_%_cpu,$(TARGETS_GCC_NVCC))
+TEST_CUDA_TARGETS := $(patsubst %,test_%_cuda,$(TARGETS_GCC_NVCC))
+TEST_SYCL_TARGETS := $(patsubst %,test_%_cuda,$(TARGETS_SYCL))
+
 all: $(TARGETS)
-test: test_cpu test_cuda
+all_gcc_nvcc: $(TARGETS_GCC_NVCC)
+all_sycl: $(TARGETS_SYCL)
+
+test: test_cpu test_cuda test_sycl
 test_cpu: $(TEST_CPU_TARGETS)
 test_cuda: $(TEST_CUDA_TARGETS)
+test_sycl: $(TEST_SYCL_TARGETS)
```

Comment:
I'm not completely happy with this as the possible `test_cpu` target from `sycltest` would be included in the `test_cpu` target here as well, therefore bringing in the dependence on SYCL toolchain.

How about using 
* the existence of `$(CUDA_BASE)` directory for `nvcc`
* `$(SYCL_BASE)` being non-empty for SYCL

as criteria for including CUDA and SYCL-dependent programs to `all` and `test_X`?